sdk/test/Services/S3/UnitTests/Custom/FailurePolicyTests.cs

```diff
@@ -0,0 +1,198 @@
+﻿using Amazon.S3;
+using Amazon.S3.Model;
+using Amazon.S3.Transfer;
+using AWSSDK.S3.Transfer.Model;
+using Amazon.S3.Transfer.Internal;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AWSSDK.UnitTests
+{
+    [TestClass]
+    public class FailurePolicyTests
+    {
+        private static TransferUtilityDownloadDirectoryRequest CreateRequest(string localDir, FailurePolicy policy)
```

Comment:
do we have a test for

1. Test what happens when user cancels mid-download with `CancellationToken`

Reply:
Added